src/api_internal.cpp

```diff
@@ -240,6 +133,58 @@ int spacemesh_api_stop(uint32_t ms_timeout)
 	return SPACEMESH_API_ERROR_NONE;
 }
 
+extern "C" int spacemesh_api_get_providers(
+	PostComputeProvider *providers, // out providers info buffer, if NULL - return count of available providers
+	int max_providers			    // buffer size
+)
+{
+	int i;
+	int current_providers = 0;
+
+	spacemesh_api_init();
+
+	if (NULL == providers) {
+		for (i = 0; i < s_total_devices; i++) {
+			if (NULL != s_gpus[i].drv && 0 != (s_gpus[i].drv->type & (SPACEMESH_API_CUDA | SPACEMESH_API_VULKAN))) {
+				current_providers++;
+			}
+		}
+		if (s_cpu.available) {
+			current_providers++;
+		}
+		return current_providers;
+	}
+
+	for (i = 0; current_providers < max_providers && i < s_total_devices; i++) {
+		if (NULL != s_gpus[i].drv && 0 != (s_gpus[i].drv->type & (SPACEMESH_API_CUDA | SPACEMESH_API_VULKAN))) {
+			if (0 != (s_gpus[i].drv->type & SPACEMESH_API_CUDA)) {
+				providers->compute_api = COMPUTE_API_CLASS_CUDA;
+			}
+			else {
+				providers->compute_api = COMPUTE_API_CLASS_VULKAN;
+			}
+			providers->id = i;
+			memcpy(providers->model, s_gpus[i].name, min(sizeof(providers->model), sizeof(s_gpus[i].name)));
+			providers->model[sizeof(providers->model) - 1] = 0;
+
+			providers++;
+			current_providers++;
+		}
+	}
+
+	if (s_cpu.available && current_providers < max_providers) {
+		providers->compute_api = COMPUTE_API_CLASS_CPU;
+		providers->id = i;
+		providers->model[0] = 'C';
+		providers->model[2] = 'P';
```

Comment:
`model[1]` is missed here, so the string isn't correct.

Reply:
Thank you!!!
Fixed.